extender.go

```diff
@@ -66,12 +69,24 @@ func (f *ExtenderFactory) NewExtender(
 		DockerfileApplier: dockerfileApplier,
 		Logger:            logger,
 	}
+	if err := f.setImageRef(extender, analyzedPath); err != nil {
+		return nil, err
+	}
 	if err := f.setExtensions(extender, groupPath, logger); err != nil {
 		return nil, err
 	}
 	return extender, nil
 }
 
+func (f *ExtenderFactory) setImageRef(extender *Extender, path string) error {
+	analyzedMD, err := f.configHandler.ReadAnalyzed(path)
+	if err != nil {
+		return err
+	}
+	extender.ImageRef = analyzedMD.BuildImage.Reference
```

Comment:
Do we need guards around any part of `BuildImage.Reference` being `nil`/`empty`?

Reply:
Thank you @jabrown85! Added in 35c052e2